src/ldp/http/AcceptPreferenceParser.ts

```diff
@@ -40,6 +41,13 @@ export class AcceptPreferenceParser extends PreferenceParser {
       result.datetime = [{ value: input.headers['accept-datetime'] as string, weight: 1 }];
     }
 
+    if (!result.type) {
+      result.type = [];
+    }
+
+    // Always reject quads as output
+    result.type.push({ value: INTERNAL_QUADS, weight: 0 });
```

Comment:
It feels a bit weird to me that adding this to the array rejects the given content type.
I may be missing a lot of background, so there may be a good reason for this.
But from an outsider's perspective, the comment and the line below seem to contradict each other.

Instead of using `0` as a magic value to represent rejection, what about having an optional boolean field such as `reject: true`? This would make things very explicit.

Reply:
This is using the same logic as the content negotiation [RFC](https://tools.ietf.org/html/rfc7231#page-38) though:

>    The weight is normalized to a real number in the range 0 through 1,
   where 0.001 is the least preferred and 1 is the most preferred; a
   value of 0 means "not acceptable".  If no "q" parameter is present,
   the default weight is 1.

Reply:
Update after discussion: add comment to the preferences interface that a weight of 0 corresponds to a format not being acceptable since not everyone knows the RFC by heart.

Reply:
> Could also add an input parameter taking forbidden content-types into account

That was my thought too, though. So feature request 🙂

Reply:
Well you're just in time since I was just going to merge 😄 . But thinking about it now, that would actually break our implementation right now since there are cases where we want the converting store to return quads. Specifically the acl handler requests quads from an acl file and does this through this same store. So if we forbid the generation of quads as a constructor parameter we would have to create an additional converting store for that case. That is also one of the reasons why I put the preference there: because it is a preference of what needs to be returned to the response.

Reply:
I find

> that would actually break our implementation right now

weird, given that having it as a parameter leaves the option of excluding it or not, whereas hardcoding it (as done now) does not leave that option 😄

However, perhaps the banlist could be applied conditionally: if no preference given for `internal/quads`, only then apply `q=0`.

Alternatively, we could just ban `internal/*` unconditionally (perhaps also excepting an explicit preference).

Reply:
By breaking I mean that we have several classes depending on converting to `internal/quads` (acl management, patching, sparql backend) so they would not be happy with a global ban on that type. With the current implementation we ban the format only in the case where it is not wanted (outputting to the response).

We could have exceptions like ignoring a 0 if we also have a 1 for that type, but that does break with the original definition of the weight I think since 0 is specifically defined as being not acceptable (and all other values are just how much preferred that type is).

Reply:
> We could have exceptions like ignoring a 0 if we also have a 1 for that type, but that does break with the original definition of the weight

Not really, since we are _adding_ to the preference string. So it's more like: if you don't say anything about internal/*, I'm assuming you don't know about it and I'll add "q=0" myself. If you know about it, on the other hand, I'm not going to add anything.

So not changing the algorithm itself.